### PR TITLE
@damassi => Switch from inf scroll to windowed pagination using cursor info

### PR DESCRIPTION
**Pull request by mzikherman**
This is the Reaction implementation for our desired pagination, using the cursor metadata provided in https://github.com/artsy/metaphysics/pull/1087 (still need to finish off that MP PR, thus the WIP here).

It seems even more straightforward than `page/size`, and is behaving quite nicely in local testing.

Little gif to try to demonstrate scrolling thru pages, and different cases (first, middle, end).

bp

bp

bp

### CODE REVIEW

File: src/Components/ArtistFilter/Artworks.tsx [hunk 1]
```diff
@@ -14,40 +9,70 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
...
+  renderPages(filters, pageCursors, options = { prefix: "", suffix: null }) {
+    return pageCursors.map(pageCursor => {
+      const { cursor, isCurrent, page } = pageCursor
+      const { prefix, suffix } = options
+      return (
+        <span
+          onClick={() => {
+            this.loadArtworks(filters, cursor, page)
+          }}
+          style={{
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Artworks.tsx]: Just something rudimentary to try to distinguish the current page...

File: src/Components/ArtistFilter/Artworks.tsx [hunk 2]
```diff
@@ -14,40 +9,70 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
...
       <div>
-        <div>Current Page: {filters.state.page}</div>
-        <div
-          onClick={() => {
-            this.loadMoreArtworks(filters)
-          }}
-        >
-          Next
+        <div>
+          {first
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Artworks.tsx]: This is the only 'display' logic in terms of, any sort of separator in the pagination bar.

File: src/Components/ArtistFilter/Artworks.tsx [hunk 3]
```diff
@@ -73,22 +98,38 @@ class Artworks extends React.Component<Props> {
...
         ) {
         __id
-        artworks: artworks_connection(first: $count, after: $cursor)
-          @connection(key: "Artworks_filtered_artworks") {
+        artworks: artworks_connection(first: $first, after: $after) {
           pageInfo {
             hasNextPage
             endCursor
           }
+          pageCursors {
```
Comment (mzikherman) [inline src/Components/ArtistFilter/Artworks.tsx]: I didn't do it in this PR, but the pagination stuff can neatly go into its own component/fragment container.

**Comment by mzikherman** [comment]
Once https://github.com/artsy/metaphysics/pull/1087 is finalized and merged, I'll remove the WIP title here.

File: src/Components/ArtistFilter/Artworks.tsx [hunk 4]
```diff
@@ -73,22 +98,38 @@ class Artworks extends React.Component<Props> {
...
+            }
+            first {
+              cursor
+              page
+              isCurrent
+            }
+            last {
+              cursor
+              page
+              isCurrent
```
Comment (alloy) [inline src/Components/ArtistFilter/Artworks.tsx]: I haven’t actually tried this with Relay Modern (I don’t think), but maybe you could use a fragment to DRY up these repetitive selections.

Comment (mzikherman) [reply src/Components/ArtistFilter/Artworks.tsx]: Yea, I mentioned above, but this definitely can just be moved into its own component and fragment container (w/ fragment there as well).

Comment (alloy) [reply src/Components/ArtistFilter/Artworks.tsx]: Oh I thought you meant the entire pagination widget.

**Comment by mzikherman** [comment]
Merging this, the MP schema stuff has been merged/deployed, and this is working fine!

It's still not terribly styled or anything, but nevertheless serves as an example of working filter w/ state (via `unstated`), and pagination using the new cursor data w/ `createRefetchContainer`.

**Pull request merged**